src/txn/txn.c

```diff
@@ -1079,21 +1085,23 @@ __wt_txn_prepare(WT_SESSION_IMPL *session, const char *cfg[])
 	/* Prepare updates. */
 	for (i = 0, op = txn->mod; i < txn->mod_count; i++, op++) {
 		/* Assert it's not an update to the lookaside file. */
-		WT_ASSERT(session,
-		    S2C(session)->cache->las_fileid == 0 ||
-		    op->fileid != S2C(session)->cache->las_fileid);
+		WT_ASSERT(session, S2C(session)->cache->las_fileid == 0 ||
+		    ((WT_BTREE *)op->dhandle->handle)->id !=
+		    S2C(session)->cache->las_fileid);
```

Comment:
I think this reads better as:
```
!F_ISSET((WT_BTREE *)op->dhandle->handle, WT_BTREE_LOOKASIDE);
```

Similarly below.

If you saved a `WT_BTREE` in the `WT_TXN_OP` structure instead of a `WT_DHANDLE` you could avoid these casts - the btree has a pointer back to the `WT_DHANDLE`.